pkg/common/config/config.go

```diff
@@ -255,23 +255,29 @@ func (cfg *Config) FromEnv() error {
 // ReadConfig parses vSphere cloud config file and stores it into VSphereConfig.
 // Environment variables are also checked
 func ReadConfig(byConfig []byte) (*Config, error) {
+	var cfg *Config
+	var err error
+
 	if len(byConfig) == 0 {
 		return nil, fmt.Errorf("Invalid YAML/INI file")
 	}
 
-	cfg, err := ReadConfigYAML(byConfig)
-	if err != nil {
-		klog.Warningf("ReadConfigYAML failed: %s", err)
-
+	// Check to see if yaml file before calling parser to prevent unnecessary error messages.
+	if isConfigYaml(byConfig) {
+		cfg, err = ReadConfigYAML(byConfig)
```

Comment:
So one of the problems is - may be the config was indeed YAML, but may be YAML was invalid. We can't assume that cfg was ini in that case. That is why IMO - if config parsing fails using YAML and then fails using INI too, then. we should log errors from both YAML and INI parsers.

Reply:
I like that idea.  I can update to have both errors printed after all attempts.  I'll mock that up and see how it goes.